### DISCUSSION TITLE
如何使用 config动态include

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#4312] [Q&A]
如何使用 config动态include呢？比如如下目录：
![image](https://github.com/xmake-io/xmake/assets/24759100/402f2977-49f7-4b84-a378-44fef17f10f4)

我希望 xmake f --demo=test1 时候脚本include(test1) , xmake f --demo=test2 时候脚本include(test2) , 甚至xmake f --demo=xxx 时候脚本include(xxx)目录 ,但是 xmake执行顺序是on_load -> after_load -> on_config -> before_build -> on_build -> after_build的，只能on_load才能动态add deps，on_config 时候已经固定了，这就没法使用这种方法了，请问还有什么方法实现？

**Comment by star-hengxing** [comment]
我觉得用 has_config + set_enabled 更好

**Comment by waruqi** [comment]
https://github.com/tboox/tbox/blob/6d19e51563d005660d3789bf218d9e376fc84b06/src/xmake.lua#L139

**Reply by PeakRacing** [reply]
这个可以
